analytics/lib/counts.py

```diff
@@ -471,71 +530,88 @@ def do_pull_minutes_active(property: str, start_time: datetime, end_time: dateti
+                  sql_data_collector(UserCount, check_realmauditlog_by_user_query(
+                      realm), (UserProfile, 'is_bot')),
+                  CountStat.DAY),
+        # Sanity check on 'active_users_audit:is_bot:day', and a archetype for future LoggingCountStats.
+        # In RealmCount, 'active_users_audit:is_bot:day' should be the partial
+        # sum sequence of 'active_users_log:is_bot:day', for any realm that
+        # started after the latter stat was introduced.
+        LoggingCountStat('active_users_log:is_bot:day',
+                         RealmCount, CountStat.DAY),
+        # Another sanity check on 'active_users_audit:is_bot:day'. Is only an
+        # approximation, e.g. if a user is deactivated between the end of the
+        # day and when this stat is run, they won't be counted. However, is the
+        # simplest of the three to inspect by hand.
+        CountStat('active_users:is_bot:day',
+                  sql_data_collector(RealmCount, count_user_by_realm_query(realm), (UserProfile, 'is_bot')),
+                  CountStat.DAY, interval=TIMEDELTA_MAX),
+
+        # User Activity stats
+        # Stats that measure user activity in the UserActivityInterval sense.
+
+        CountStat('1day_actives::day',
+                  sql_data_collector(
+                      UserCount, check_useractivityinterval_by_user_query(realm), None),
+                  CountStat.DAY, interval=timedelta(days=1)-UserActivityInterval.MIN_INTERVAL_LENGTH),
+        CountStat('15day_actives::day',
+                  sql_data_collector(
+                      UserCount, check_useractivityinterval_by_user_query(realm), None),
+                  CountStat.DAY, interval=timedelta(days=15)-UserActivityInterval.MIN_INTERVAL_LENGTH),
+        CountStat('minutes_active::day', DataCollector(
+            UserCount, do_pull_minutes_active), CountStat.DAY),
+
+        # Rate limiting stats
+
+        # Used to limit the number of invitation emails sent by a realm
+        LoggingCountStat('invites_sent::day', RealmCount, CountStat.DAY),
+
+        # Dependent stats
+        # Must come after their dependencies.
+
+        # Canonical account of the number of active humans in a realm on each day.
+        DependentCountStat('realm_active_humans::day',
+                           sql_data_collector(
+                               RealmCount, count_realm_active_humans_query(realm), None),
+                           CountStat.DAY,
+                           dependencies=['active_users_audit:is_bot:day', '15day_actives::day'])
+    ]
+    return count_stats_
+
+# To avoid refactoring for now COUNT_STATS can be used as before
+count_stats_ = get_count_stats_()
 COUNT_STATS = OrderedDict([(stat.property, stat) for stat in count_stats_])
+
+def get_count_stats(realm: Optional[Realm]=None) -> Dict[str, CountStat]:
+    count_stats_ = get_count_stats_(realm)
+    return OrderedDict([(stat.property, stat) for stat in count_stats_])
```

Comment:
Why do you need the `get_count_stats_` function, rather than e.g. having `get_count_stats` just return an OrderedDict if all the callers want that format?

Reply:
@timabbott this was due to requirement for the list of COUNT_STATS objects in future but its highly unlikely that it will be needed. So pushed the cleaner code.

Reply:
Yeah, I think making code uglier to simplify support for an unlikely potential future use case is a pattern we want to avoid wherever we can.